### DISCUSSION TITLE
Should I install hardhat globally?

### DISCUSSION THREAD

**Discussion by CaledoniaProject** [NomicFoundation/hardhat#2955] [General]
I'm wondering if I should I install hardhat globally? I installed truffle with `-g`, I shouldn't install hardhat for each project right?

**Comment by fvictorio** [comment]
On the contrary, Hardhat doesn't even work if you install it globally. This is an explicit decision on our part since we think installing tools like this is an anti-pattern, given that it reduces reproducibility among members of the team.

Notice that this is a very common recommendation. For example, from eslint's docs:

> It is also possible to install ESLint globally rather than locally (using npm install eslint --global). However, this is not recommended, and any plugins or shareable configs that you use must be installed locally in either case.

Or check [Prettier's installation docs](https://prettier.io/docs/en/install.html), where the possibility of installing it globally is not even mentioned (even if prettier does work in that case).
